### ISSUE TITLE
Issue when reusing variables with trap/panic

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#19099]
**Description:**
```ballerina
import ballerina/io;

public function main() {
    var result = trap panicNow();
    io:println("one: ", result);

    result = trap dontPanic();
    io:println("two: ", result);
}

function panicNow() {
    io:println("panicNow");
    panic error("too bad!");
}

function dontPanic() {
    io:println("dontPanic");
}
```

Running this produces
```cmd
$ ballerina run abc.bal 
panicNow
one: error too bad! 
two: error too bad! 
```

`dontPanic()` is not even executed and `result` holds the error from the `panicNow()` invocation.

Only seems to happen when the first function panics, and trap is used.

**Affected Versions:**
```cmd
$ ballerina -v
Ballerina 1.0.0
Language specification 2019R3
```

**Comment by riyafa** [comment]
I noticed this occurred (in a different case) when we used the same strand to set return values. If we set the return values for a function to the strand and did not set it to null if it doesn't return anything in the next we saw it returned the earlier return value. Maybe it might be the same issue with the trap implementation. Maybe a different one, but this could be a pointer.

**Comment by warunalakshitha** [comment]
Fixed 1.0.1
